control-plane/connect-inject/controllers/pod/pod_controller.go

```diff
@@ -811,11 +811,12 @@ func getHealthStatusFromPod(pod corev1.Pod) pbcatalog.Health {
 	}
 
 	for _, condition := range pod.Status.Conditions {
-		if condition.Type == corev1.PodReady && condition.Status == corev1.ConditionTrue {
-			return pbcatalog.Health_HEALTH_PASSING
+		if condition.Type == corev1.PodReady && condition.Status == corev1.ConditionFalse {
+			return pbcatalog.Health_HEALTH_CRITICAL
 		}
 	}
-	return pbcatalog.Health_HEALTH_CRITICAL
+
+	return pbcatalog.Health_HEALTH_PASSING
```

Comment:
Wouldn't it mean that if pod is not ready we'll return a passing health status? Also, are there unit tests for this that should be updated?

Reply:
We should return in line 815 if any of the ready conditions fail

I don't think there are unit tests but I can certainly add some for this function now!

Reply:
I'm also curious, are you seeing multiple pod ready status conditions per container? There should be only one per pod, right? So I'm curious why we needed that change.

Reply:
ya it's iterating through these pod.Status.Conditions here (this is the multiport pod during the acceptance test when we mark it unhealthy), so we're looking at anything with type: ready and then what the status is:
```
status:
  conditions:
  - lastProbeTime: null
    lastTransitionTime: "2023-09-21T20:13:01Z"
    status: "True"
    type: Initialized
  - lastProbeTime: null
    lastTransitionTime: "2023-09-21T20:13:37Z"
    message: 'containers with unready status: [multiport multiport-admin]'
    reason: ContainersNotReady
    status: "False"
    type: Ready
  - lastProbeTime: null
    lastTransitionTime: "2023-09-21T20:13:37Z"
    message: 'containers with unready status: [multiport multiport-admin]'
    reason: ContainersNotReady
    status: "False"
    type: ContainersReady
  - lastProbeTime: null
    lastTransitionTime: "2023-09-21T20:12:54Z"
    status: "True"
    type: PodScheduled

```

Reply:
on the unit tests- it does look like theres one failure that i need to fix so there might be some coverage already actually! will go back and update

Reply:
>We should return in line 815 if any of the ready conditions fail 

What if the pod doesn't have a ready condition yet for some reason?

>ya it's iterating through these pod.Status.Conditions here (this is the multiport pod during the acceptance test when we mark it unhealthy), so we're looking at anything with type: ready and then what the status is:

Hmm so there's only one Ready type condition which is false. Does the old code return that the health status is passing in this case? I'd expect it to return critical because it would iterate through all the conditions and would find none that are both type `Ready` and status `true` and so it'd return status critical by default. Maybe I'm missing something, but a unit test would probably help just to make sure we're covering all cases correctly.

Reply:
Ah I see what youre saying, I didn't realize there is only one ready status. let me try testing with just the consul fix and not the k8s changes here, and see if it still works. In either case, I can add unit tests

Reply:
I've reverted my changes to this function since it was correct in the first place. Thanks for catching @ishustava!

I'm seeing there is full unit test coverage of the health status behavior, its tested in TestHealthStatusWrite which calls out to writeHealthStatus which calls getHealthStatusFromPod. It checks all 3 cases- when ready condition is true, false, and nonexistent, and the logic is correct there. 

I didn't look closely enough at the health statuses to realize there was only one ready status, and the test passes without changes to pod controller and just the consul fixes!